### ISSUE TITLE
Callback broken by use of Spinner

### ISSUE THREAD

**Issue by mdk73** [kivy/kivy#3855]
I create small modules (and kv-files) that build on each other to step-wise enhance functionality.
While the single modules are working, importing them in other modules has unintended side effects.
Especially I have problems using callbacks. Sometimes they work, sometimes not. 
And sometimes simply adding a widget breaks the callback.

The following testcase shows how adding a spinner to the kv-file breaks a callback.
The importtest-files (py and kv) are given as well, they are not changed in this test case.
I know, the test case should rather contain less code, but somehow the imported (and not really used) code influences the behaviour of the callback. I'll show that in a separate issue.

I start with a working code example. A callback is attached to the collapse property of a accordion item. Clicking on the accordion items prints the status 'False' or 'True', as shown in the output below.
```python
# test.py
from importtest import ImportTest
from kivy.app import App
from kivy.uix.widget import Widget

class Test(Widget):
    def __init__(self, **kwargs):
        super(Test, self).__init__(**kwargs)
        self.importtest = ImportTest()
        self.item_1.bind(collapse = self.on_collapse)
        
    def on_collapse(self, instance, value):
        print 'on_collapse'
        if value is True:
            print 'True'
        else:
            print 'False'        

class TestApp(App):
    def build(self):
        return Test().sm

if __name__ == '__main__':
    TestApp().run()
```
```kivy
# test.kv     
<Test>:
    sm: sm
    item_1: item_1
    item_2: item_2
    
    ScreenManager:
        id: sm
        Screen:
#             Splitter:
#                 sizable_from: 'right'
#                 min_size: 10
            Accordion:
                orientation: 'horizontal'
                AccordionItem:
                    id: item_1
                    title: '1'
                AccordionItem:
                    id: item_2
                    title: '2'
```

```python
# importtest.py
from kivy.app import App
from kivy.lang import Builder
from kivy.uix.boxlayout import BoxLayout
from kivy.uix.button import Button
from kivy.uix.dropdown import DropDown
from kivy.uix.treeview import TreeView

if __name__ != '__main__':
    Builder.load_file('importtest.kv')

class MyTreeView(TreeView):
    def __init__(self, **kwargs):
        super(MyTreeView, self).__init__(**kwargs)
        
class ImportTest(BoxLayout):  
    def __init__(self, **kwargs):
        super(ImportTest, self).__init__(**kwargs)
        self.treeview = MyTreeView()
        
class ImportTestApp(App):
    def build(self):
        return ImportTest()
```
```kivy
# importtest.kv
<ImportTest>:
    sv: sv
    layout: layout
    box_layout: box_layout
    orientation: 'horizontal'
    size_hint: (1,1)
             
    BoxLayout:
        id: box_layout
        orientation: 'vertical'
        pos: root.pos
        size: root.size
         
        ScrollView:
            pos_hint: {'top': 1}
            id: sv
            size_hint: (1, 1)
             
            BoxLayout:
                id: layout
                size_hint_y: None
                orientation: 'vertical'
        
<MyTreeView>:
    size_hint_y: None
```
Output:
```
[INFO              ] [Logger      ] Record log in C:\Users\mdk\.kivy\logs\kivy_15-12-20_113.txt
[INFO              ] [Kivy        ] v1.9.1-dev
[INFO              ] [Python      ] v2.7.11 (v2.7.11:6d1b6a68f775, Dec  5 2015, 20:40:30) [MSC v.1500 64 bit (
AMD64)]
[INFO              ] [Factory     ] 179 symbols loaded
[INFO              ] [Image       ] Providers: img_tex, img_dds, img_gif, img_sdl2, img_pil (img_ffpyplayer ig
nored)
[INFO              ] [Text        ] Provider: sdl2
[INFO              ] [OSC         ] using <thread> for socket
[INFO              ] [Window      ] Provider: sdl2
[INFO              ] [GL          ] GLEW initialization succeeded
[INFO              ] [GL          ] OpenGL version <4.5.0 NVIDIA 353.62>
[INFO              ] [GL          ] OpenGL vendor <NVIDIA Corporation>
[INFO              ] [GL          ] OpenGL renderer <Quadro 1000M/PCIe/SSE2>
[INFO              ] [GL          ] OpenGL parsed version: 4, 5
[INFO              ] [GL          ] Shading version <4.50 NVIDIA>
[INFO              ] [GL          ] Texture max size <16384>
[INFO              ] [GL          ] Texture max units <32>
[INFO              ] [Window      ] auto add sdl2 input provider
[INFO              ] [Window      ] virtual keyboard not allowed, single mode, not docked
[INFO              ] [Base        ] Start application main loop
[INFO              ] [GL          ] NPOT texture support is available
on_collapse
False
on_collapse
True
[INFO              ] [Base        ] Leaving application in progress...
```

Uncommenting the Spinner and indenting the Accordion in the kv-file breaks the callback. When the accordion items are clicked it works only once, then not any more.
```kivy
# test.kv - version 2   
<Test>:
    sm: sm
    item_1: item_1
    item_2: item_2
    
    ScreenManager:
        id: sm
        Screen:
            Splitter:
                sizable_from: 'right'
                min_size: 10
                Accordion:
                    orientation: 'horizontal'
                    AccordionItem:
                        id: item_1
                        title: '1'
                    AccordionItem:
                        id: item_2
                        title: '2'
```
Output:
```
[INFO              ] [Logger      ] Record log in C:\Users\mdk\.kivy\logs\kivy_15-12-20_114.txt
[INFO              ] [Kivy        ] v1.9.1-dev
[INFO              ] [Python      ] v2.7.11 (v2.7.11:6d1b6a68f775, Dec  5 2015, 20:40:30) [MSC v.1500 64 bit (
AMD64)]
[INFO              ] [Factory     ] 179 symbols loaded
[INFO              ] [Image       ] Providers: img_tex, img_dds, img_gif, img_sdl2, img_pil (img_ffpyplayer ig
nored)
[INFO              ] [Text        ] Provider: sdl2
[INFO              ] [OSC         ] using <thread> for socket
[INFO              ] [Window      ] Provider: sdl2
[INFO              ] [GL          ] GLEW initialization succeeded
[INFO              ] [GL          ] OpenGL version <4.5.0 NVIDIA 353.62>
[INFO              ] [GL          ] OpenGL vendor <NVIDIA Corporation>
[INFO              ] [GL          ] OpenGL renderer <Quadro 1000M/PCIe/SSE2>
[INFO              ] [GL          ] OpenGL parsed version: 4, 5
[INFO              ] [GL          ] Shading version <4.50 NVIDIA>
[INFO              ] [GL          ] Texture max size <16384>
[INFO              ] [GL          ] Texture max units <32>
[INFO              ] [Window      ] auto add sdl2 input provider
[INFO              ] [Window      ] virtual keyboard not allowed, single mode, not docked
[INFO              ] [Base        ] Start application main loop
[INFO              ] [GL          ] NPOT texture support is available
on_collapse
False
[INFO              ] [Base        ] Leaving application in progress...
```

**Comment by bionoid** [comment]
Probably the kv rules are not finished processing by the time you create the binding in ``__init__``.  Try ``on_collapse: root.callback(self)`` in your AccordionItem, or add a new method to do the actual bind and offset it to the next frame using something like ``Clock.schedule_once(self.do_bind)`` in ``__init__``

(I didn't actually try the code though)

**Comment by mdk73** [comment]
This variant `on_collapse: root.on_collapse(*args)` gives me the following traceback:
```
[INFO              ] [Logger      ] Record log in C:\Users\mdk\.kivy\logs\kivy_15-12-20_122.txt
[INFO              ] [Kivy        ] v1.9.1-dev
[INFO              ] [Python      ] v2.7.11 (v2.7.11:6d1b6a68f775, Dec  5 2015, 20:40:30) [MSC v.1500 64 bit (
AMD64)]
[INFO              ] [Factory     ] 179 symbols loaded
[INFO              ] [Image       ] Providers: img_tex, img_dds, img_gif, img_sdl2, img_pil (img_ffpyplayer ig
nored)
[INFO              ] [Text        ] Provider: sdl2
[INFO              ] [OSC         ] using <thread> for socket
[INFO              ] [Window      ] Provider: sdl2
[INFO              ] [GL          ] GLEW initialization succeeded
[INFO              ] [GL          ] OpenGL version <4.5.0 NVIDIA 353.62>
[INFO              ] [GL          ] OpenGL vendor <NVIDIA Corporation>
[INFO              ] [GL          ] OpenGL renderer <Quadro 1000M/PCIe/SSE2>
[INFO              ] [GL          ] OpenGL parsed version: 4, 5
[INFO              ] [GL          ] Shading version <4.50 NVIDIA>
[INFO              ] [GL          ] Texture max size <16384>
[INFO              ] [GL          ] Texture max units <32>
[INFO              ] [Window      ] auto add sdl2 input provider
[INFO              ] [Window      ] virtual keyboard not allowed, single mode, not docked
[INFO              ] [Base        ] Start application main loop
[INFO              ] [GL          ] NPOT texture support is available
on_collapse
False
[INFO              ] [Base        ] Leaving application in progress...
 Traceback (most recent call last):
   File "C:\Users\mdk\workspace\lr_control\src\gui\test.py", line 24, in <module>
     TestApp().run()
   File "C:\Python27\lib\site-packages\kivy\app.py", line 828, in run
     runTouchApp()
   File "C:\Python27\lib\site-packages\kivy\base.py", line 487, in runTouchApp
     EventLoop.window.mainloop()
   File "C:\Python27\lib\site-packages\kivy\core\window\window_sdl2.py", line 606, in mainloop
     self._mainloop()
   File "C:\Python27\lib\site-packages\kivy\core\window\window_sdl2.py", line 361, in _mainloop
     EventLoop.idle()
   File "C:\Python27\lib\site-packages\kivy\base.py", line 330, in idle
     self.dispatch_input()
   File "C:\Python27\lib\site-packages\kivy\base.py", line 315, in dispatch_input
     post_dispatch_input(*pop(0))
   File "C:\Python27\lib\site-packages\kivy\base.py", line 221, in post_dispatch_input
     listener.dispatch('on_motion', etype, me)
   File "kivy\_event.pyx", line 718, in kivy._event.EventDispatcher.dispatch (kivy\_event.c:7699)
   File "C:\Python27\lib\site-packages\kivy\core\window\__init__.py", line 993, in on_motion
     self.dispatch('on_touch_down', me)
   File "kivy\_event.pyx", line 718, in kivy._event.EventDispatcher.dispatch (kivy\_event.c:7699)
   File "C:\Python27\lib\site-packages\kivy\core\window\__init__.py", line 1009, in on_touch_down
     if w.dispatch('on_touch_down', touch):
   File "kivy\_event.pyx", line 718, in kivy._event.EventDispatcher.dispatch (kivy\_event.c:7699)
   File "C:\Python27\lib\site-packages\kivy\uix\screenmanager.py", line 1070, in on_touch_down
     return super(ScreenManager, self).on_touch_down(touch)
   File "C:\Python27\lib\site-packages\kivy\uix\widget.py", line 432, in on_touch_down
     if child.dispatch('on_touch_down', touch):
   File "kivy\_event.pyx", line 718, in kivy._event.EventDispatcher.dispatch (kivy\_event.c:7699)
   File "C:\Python27\lib\site-packages\kivy\uix\relativelayout.py", line 278, in on_touch_down
     ret = super(RelativeLayout, self).on_touch_down(touch)
   File "C:\Python27\lib\site-packages\kivy\uix\widget.py", line 432, in on_touch_down
     if child.dispatch('on_touch_down', touch):
   File "kivy\_event.pyx", line 718, in kivy._event.EventDispatcher.dispatch (kivy\_event.c:7699)
   File "C:\Python27\lib\site-packages\kivy\uix\widget.py", line 432, in on_touch_down
     if child.dispatch('on_touch_down', touch):
   File "kivy\_event.pyx", line 718, in kivy._event.EventDispatcher.dispatch (kivy\_event.c:7699)
   File "C:\Python27\lib\site-packages\kivy\uix\widget.py", line 432, in on_touch_down
     if child.dispatch('on_touch_down', touch):
   File "kivy\_event.pyx", line 718, in kivy._event.EventDispatcher.dispatch (kivy\_event.c:7699)
   File "C:\Python27\lib\site-packages\kivy\uix\accordion.py", line 302, in on_touch_down
     self.collapse = False
   File "kivy\properties.pyx", line 408, in kivy.properties.Property.__set__ (kivy\properties.c:5114)
   File "kivy\properties.pyx", line 446, in kivy.properties.Property.set (kivy\properties.c:5876)
   File "kivy\properties.pyx", line 501, in kivy.properties.Property.dispatch (kivy\properties.c:6557)
   File "kivy\_event.pyx", line 1224, in kivy._event.EventObservers.dispatch (kivy\_event.c:13497)
   File "kivy\_event.pyx", line 1130, in kivy._event.EventObservers._dispatch (kivy\_event.c:12696)
   File "C:\Python27\lib\site-packages\kivy\uix\accordion.py", line 282, in on_collapse
     self.accordion.select(self)
   File "C:\Python27\lib\site-packages\kivy\uix\accordion.py", line 384, in select
     widget.collapse = widget is not instance
   File "kivy\properties.pyx", line 408, in kivy.properties.Property.__set__ (kivy\properties.c:5114)
   File "kivy\properties.pyx", line 446, in kivy.properties.Property.set (kivy\properties.c:5876)
   File "kivy\properties.pyx", line 501, in kivy.properties.Property.dispatch (kivy\properties.c:6557)
   File "kivy\_event.pyx", line 1224, in kivy._event.EventObservers.dispatch (kivy\_event.c:13497)
   File "kivy\_event.pyx", line 1106, in kivy._event.EventObservers._dispatch (kivy\_event.c:12301)
   File "C:\Python27\lib\site-packages\kivy\lang.py", line 1557, in custom_callback
     exec(__kvlang__.co_value, idmap)
   File "C:\Users\mdk\workspace\lr_control\src\gui\test.kv", line 18, in <module>
     on_collapse: root.on_collapse(*args)
   File "kivy\weakproxy.pyx", line 19, in kivy.weakproxy.WeakProxy.__getattr__ (kivy\weakproxy.c:1143)
   File "kivy\weakproxy.pyx", line 15, in kivy.weakproxy.WeakProxy.__ref__ (kivy\weakproxy.c:1042)
 ReferenceError: weakly-referenced object no longer exists
```

**Comment by mdk73** [comment]
@bionoid :
Your suggestion with separate methods and clock.schedule might work. 
But I am rather hesitatingly, this does not seem like a very "pythonic" way. 
Unless I made mistakes in my syntax, I think it would be much better if the code would work reliable, undisturbed by added / removed Widgets or imports. 
Am I wrong?

**Comment by dessant** [comment]
@bionoid's suggestion should work. PR https://github.com/kivy/kivy/pull/3456 will introduce more intuitive ways of taking care of this.
